vignettes/function-flow.Rmd

```diff
@@ -95,63 +95,202 @@ outputs.](images/function-flow-population.png)
+    -   `is_t1d`: Define whether the diagnosis was T1D-specific
+        (`diagnosekode` starts with "DE10")
+    -   `is_t2d`: Define whether the diagnosis was T2D-specific
+        (`diagnosekode` starts with "DE11")
+    -   `department`: Define whether the diagnosis was made made by an
+        endocrinological (`hovedspeciale_ans` == "medicinsk
+        endokrinologi") or other medical department (`hovedspeciale_ans`
+        either "Blandet medicin og kirurgi", "Intern medicin",
+        "Geriatri", "Hepatologi", "Hæmatologi", "Infektionsmedicin",
+        "Kardiologi", "Medicinsk allergologi", "Medicinsk
+        gastroenterologi", "Medicinsk lungesygdomme", "Nefrologi",
+        "Reumatologi", "Palliativ medicin", "Akut medicin",
+        "Dermato-venerologi", "Neurologi", "Onkologi", "Fysiurgi", or
+        "Tropemedicin")
+
+These intermediate results are combined for further processing, and
+`include_diabetes_diagnoses()` outputs a single `data.frame` with the
+following variables (one row for each individual):
+
+-   identifier variable (`pnr`)
+-   date of the first diabetes diagnosis (`do_diagnosis_1`)
+-   date of the second diabetes diagnosis (`do_diagnosis_2`)
+-   number of type 1 diabetes-specific primary diagnosis codes from
+    endocrinological departments (`n_t1d_endo`)
+-   number of type 2 diabetes-specific primary diagnosis codes from
+    endocrinological departments (`n_t2d_endo`)
+-   number of type 1 diabetes-specific primary diagnosis codes from
+    medical departments (`n_t1d_medical`)
+-   number of type 2 diabetes-specific primary diagnosis codes from
+    medical departments (`n_t2d_medical`)
+
+The output is passed to the `get_diagnosis_date()` function for the
+final step of the inclusion process and is subsequently used to classify
+diabetes type.
 
 #### Diabetes-specific podiatrist services
 
 The function `include_podiatrist_services()` uses `sysi` or `sssy` as
 input to extract the dates of all diabetes-specific podiatrist services.
 
-<!-- TODO: Add details on how this filtering should be done -->
+These dates are extracted by filtering values beginning with "54" in the
+`spec` variable of the `sssy` and `sysi` registers by default
+(alternatively, the function can take the `spec2` variable as input
+instead, if that is the data available to the user). In addition,
+services provided to a child of the individual (`barnmak` != 0) are
+excluded using the `barnmak` variable. An internal helper function
+`get_unique_honuge_dates()` is applied to generate a proper date
+variable based on the year-week (wwyy-formatted) variable (`honuge`)
+found in the raw data, and de-duplicates multiple services registered on
+the same date.
+
+`include_podiatrist_services()` outputs a 3-column data frame with one
+row for each individual, containing the following variables:
+
+-   identifier variable (`pnr`)
+-   the date of the first diabetes-specific podiatrist record
+    (`do_podiatrist_1`)
```

Comment:
What does this `do_` mean? If date, let's call it `date_`.

Reply:
Yes, or as we discussed: use a longer format with just one variable for all dates.